### ISSUE TITLE
Provide import/export feature

### ISSUE THREAD

**Issue by maulwuff** [dmfs/opentasks#170]
It would be nice to export tasks and import them into another task list.
Furthermore, sharing a task to android would be nice (send as email etc)

**Comment by dmfs** [comment]
Please create a separate ticket for sharing since that's a different thing.

**Comment by maulwuff** [comment]
Ok, it's in #172

**Comment by marcelklehr** [comment]
+1

**Comment by jkufner** [comment]
I do not really see reason why to export tasks. They are on your CalDAV server and you can fetch them form there in ICS format with no trouble. 

Moving task from one calendar to another is very useful feature as well as sharing, but they are both something different than import/export.

**Comment by jkufner** [comment]
I created issue #198 about moving tasks from one calendar to another.

**Comment by marcelklehr** [comment]
If you have your tasks on a different medium (in my case I created them
with a script that ouputs ics files), it's useful to be able to import
them. Owncloud doesn't allow importing, neither does this project,
sadly. Luckily thnderbird allows importing ics files, so that's that.

IMO, this app shouldn't require you to have a calDAV server to provide
essential features, even though that's a pretty common setup, you can't
presume *the server already does that*. That's my two cents.

**Comment by mkroehnert** [comment]
@marcelklehr OwnCloud does allow importing .ics files, although it is kind of hidden.
You have to upload the .ics file to OwnCloud and then click on it in the files view.
This will open an import dialog for this specific .ics file.

**Comment by wweich** [comment]
I use the script from here to import my automatically created tasks ics file to owncloud:
https://github.com/georgehrke/cl-calendarimport

**Comment by dmfs** [comment]
Although in general it's preferable to perform imports and exports on a server, we still plan to implement this for users who use the app standalone, without syncing.

**Comment by maulwuff** [comment]
I appreciate this @dmfs 
Some notes why I prefer this feature:
- not all programs produce the same quality of ics... 
- not everybody has access to the server directly (furthermore, I don't see why this should be done on the server side, this is a client job for me.)
- not everybody uses owncloud.

**Comment by jkufner** [comment]
> not everybody has access to the server directly (furthermore, I don't see why this should be done on the server side, this is a client job for me.)

Yes. And you can simply download everything as a client without any special access on server. See http://stackoverflow.com/questions/23030628/how-to-download-all-my-caldav-and-carddav-data-with-one-wget-curl

> not everybody uses owncloud.

It does not matter at all.

**Comment by dmfs** [comment]
>> not everybody has access to the server directly (furthermore, I don't see why this should be done on the server side, this is a client job for me.)

>Yes. And you can simply download everything as a client without any special access on server. See http://stackoverflow.com/questions/23030628/how-to-download-all-my-caldav-and-carddav-data-with-one-wget-curl

Well, as "hnh" has pointed out in his response, this is not covered by the CalDAV standard. Some servers support that, but it's not required. Also, this doesn't cover the "import" use case. There are servers that can import ics files when you address a `PUT` request to a calendar collection, but again, that's not part of the specs.

**Comment by bambamal** [comment]
I found this two year old issue while looking for an export solution, is there any update on this?

I know a lot of people use this app to sync with a CalDAV server. But OpenTasks also provides local task lists, which I am sure at least a few people use (like me). If you do that, you need to be able to export and import ics files to backup your data.

**Comment by balisujohn** [comment]
I agree with @bambamal , this is a feature I personally would benefit from. Would this feature conflict with the design philosophy of this project, or is there another reason it has not been implemented?

**Comment by dmfs** [comment]
@balisujohn there is no conflict. The problem is simply higher priorities for other issues and little time to tackle them all in a timely manner.

**Comment by xCaradhras** [comment]
Hello,
I was unfortunately too lazy to set up my own server so I used the local task list feature. Now I got a new phone. How do I get my tasks to my new phone? (or server if I have to set up one)

Even if there is no "official way", is there a hacky way? For example copy file.xyz from old phone to new phone or something?

**Comment by maulwuff** [comment]
> 
> 
> Hello,
> I was unfortunately too lazy to set up my own server so I used the local task list feature. Now I got a new phone. How do I get my tasks to my new phone? (or server if I have to set up one)
> 
> Even if there is no "official way", is there a hacky way? For example copy file.xyz from old phone to new phone or something?

If you have a server now, you can add the account to your old phone, start OpenTasks and change the "calendar" for each task by editing it. At least this works for me to switch from one calendar to another. 
As for the server: as this is a companion app to CalDavSync, maybe try one of the listed providers when using CalDavSync?
